kafka-client/src/main/java/dev/responsive/kafka/config/ResponsiveConfig.java

```diff
@@ -252,13 +258,35 @@ public static ResponsiveConfig responsiveConfig(final Map<?, ?> originals) {
    * mid-application, for example during state store init.
    */
   public static ResponsiveConfig loggedConfig(final Map<?, ?> originals) {
+
+    final Properties props = loadVersionPropertiesFromFile();
+    final String version = props.getProperty("git.build.version", "").trim();
```

Comment:
is it possible to create constants here that we can share with the kotlin script?

Reply:
After much digging around on this question, I have to conclude that the answer is "no" -- at least, it's not possible to create/share constants in the way you mean or would make sense here. My understanding is that the only way to "share" constants between the gradle/kotlin scripts and our compiled java classes is basically to pass them from gradle to the JVM via environment variables or properties files. That doesn't really help us keep them in sync because we still need to know the format/name of the env/file property -- which is basically exactly what these constants are used for in the first place. ie we'd just be adding another layer of equally fragile strings to keep track of 

There do seem to be some 3rd party plugins that _might_ achieve this, but the whole reason for doing this manually was to avoid those in the first place (not to mention, they probably are just syntactic sugar over basically the same method of passing around constants 🤷‍♀️ )

Reply:
All that said, I'm still relatively a complete noob to kotlin for gradle, so I'd be very happy to be proven wrong and learn how to do this. If nothing else this whole patch has been an excellent crash course in kotlin-based gradle build scripts, so I appreciate the question even if I couldn't work out a way to do it that made sense